suite-common/wallet-core/src/send/sendFormEthereumThunks.ts

```diff
@@ -151,7 +151,14 @@ export const composeEthereumSendFormTransactionThunk = createThunk(
         } else {
             customFeeLimit = tokenInfo ? ERC20_BACKUP_GAS_LIMIT : ETH_BACKUP_GAS_LIMIT;
 
-            // TODO: catch error from blockbook/geth (invalid contract, not enough balance...)
+            dispatch(
+                notificationsActions.addToast({
+                    type: 'estimated-fee-error',
+                    error: estimatedFee.payload.error,
+                }),
+            );
+
+            return;
```

Comment:
Hmm now I'm thinking about it (because of the failed tests) we should not put a return here. We need the function to continue. If it fails the only thing that will be different is that it will use the backup value and show toast message.

I would suggest one thing for the toast. Let's not even show the error. It can be just a copy with e. g. "Failed to estimate fee from the network. Using backup value.".

Please update it like this. When you will be in the `suite-common/toast-notifications/src/types.ts` you can also do the same type of refactoring for objects with just type property (put them together with your type).

Reply:
Yes, the return should not be there. Fallback of gas limit is set on line 152